compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp

```diff
@@ -1414,13 +1418,13 @@ LLVMRustPrepareThinLTOInternalize(const LLVMRustThinLTOData *Data,
   return true;
 }
 
-extern "C" bool LLVMRustPrepareThinLTOImport(const LLVMRustThinLTOData *Data,
+extern "C" bool LLVMRustPrepareThinLTOImport(LLVMRustThinLTOData *Data,
```

Comment:
I think the problem here is that this `const` removal did not get propagated to `ffi.rs` -- if it did, we'd probably get a borrow checker error :)

It looks like this function is called from optimize_thin_module, which I believe is run in parallel, so we might end up resizing the map while another thread holds a reference to it.

Reply:
I think we can't really fix this without LLVM exposing an additional API. I left a comment on the PR at https://github.com/llvm/llvm-project/pull/106427#issuecomment-2354783802.